pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -334,3 +264,90 @@ func pluginsNeeded(plugins *config.Plugins) map[string]struct{} {
 
 	return pgMap
 }
+
+func (f *framework) CombinePlugin(plugins *config.Plugins) error {
+	if plugins == nil {
+		return nil
+	}
+	commonfn := func(es string, name string) error {
+		extendErr := fmt.Errorf("plugin %v does not extend %s plugin", name, es)
+
+		if pg, ok := f.plugins[name]; ok {
+			switch es {
+			case "reserve":
+				p, ok := pg.(ReservePlugin)
+				if !ok {
+					return extendErr
+				}
+				f.reservePlugins = append(f.reservePlugins, p)
+			case "prebind":
+				p, ok := pg.(PrebindPlugin)
+				if !ok {
+					return extendErr
+				}
+				f.prebindPlugins = append(f.prebindPlugins, p)
+			case "unreserve":
+				p, ok := pg.(UnreservePlugin)
+				if !ok {
+					return extendErr
+				}
+				f.unreservePlugins = append(f.unreservePlugins, p)
+			case "permit":
+				p, ok := pg.(PermitPlugin)
+				if !ok {
+					return extendErr
+				}
+				f.permitPlugins = append(f.permitPlugins, p)
+			case "queueSort":
+				p, ok := pg.(QueueSortPlugin)
+				if !ok {
+					name = "queue sort"
```

Comment:
why reset the name here?

Reply:
I think ``queue sort`` is better than ``queueSort`` in the error message, but that is not necessary. Removed, thanks!!